src/main/java/com/google/sps/servlets/LoadInterviewsServlet.java

```diff
@@ -0,0 +1,256 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.common.collect.ImmutableList;
+import com.google.common.base.Preconditions;
+import com.google.gson.Gson;
+import com.google.sps.data.Availability;
+import com.google.sps.data.AvailabilityDao;
+import com.google.sps.data.DatastoreAvailabilityDao;
+import com.google.sps.data.DatastoreScheduledInterviewDao;
+import com.google.sps.data.PossibleInterviewSlot;
+import com.google.sps.data.ScheduledInterview;
+import com.google.sps.data.ScheduledInterviewDao;
+import com.google.sps.data.TimeRange;
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.lang.Integer;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.TextStyle;
+import java.util.Locale;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+import javax.servlet.RequestDispatcher;
+import java.util.Optional;
+
+@WebServlet("/load-interviews")
+public class LoadInterviewsServlet extends HttpServlet {
+
+  private AvailabilityDao availabilityDao;
+  private ScheduledInterviewDao scheduledInterviewDao;
+  private Instant instant;
+
+  @Override
+  public void init() {
+    init(new DatastoreAvailabilityDao(), new DatastoreScheduledInterviewDao(), Instant.now());
+  }
+
+  public void init(
+      AvailabilityDao availabilityDao,
+      ScheduledInterviewDao scheduledInterviewDao,
+      Instant instant) {
+    this.availabilityDao = availabilityDao;
+    this.scheduledInterviewDao = scheduledInterviewDao;
+    this.instant = instant;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int timezoneOffsetMinutes = Integer.parseInt(request.getParameter("timeZoneOffset"));
+    Preconditions.checkArgument(
+        Math.abs(timezoneOffsetMinutes) <= 720,
+        "Offset greater than 720 minutes (12 hours): %s",
+        timezoneOffsetMinutes);
+    ZoneOffset timezoneOffset = convertIntToOffset(timezoneOffsetMinutes);
+    ZonedDateTime day = generateDay(instant, timezoneOffset);
+    ZonedDateTime utcTime = day.withZoneSameInstant(ZoneOffset.UTC);
+    // TODO: Decide how precise this range should be. Must start after or at now, when should end
+    // be?
+    TimeRange range =
```

Comment:
"range"?

interviewTimeRange? or interviewTime?

(I think "interviewRange" doesn't give any intuition, so i skipped that option)

Reply:
Changed to interviewSearchTimeRange